src/symbolic/solver.ml

```diff
@@ -16,10 +16,14 @@ let fresh solver () =
 
 let check (S (solver_module, s)) pc =
   let module Solver = (val solver_module) in
+  (* TODO: maybe we should change the API of smtml so that Solver.check expects a set instead of a list ? (to avoid duplicates) *)
+  let pc = Smtml.Expr.Set.to_list pc in
   Solver.check s pc
 
 let model (S (solver_module, s)) ~symbols ~pc =
   let module Solver = (val solver_module) in
+  (* TODO: maybe we should change the API of smtml so that Solver.check expects a set instead of a list ? (to avoid duplicates) *)
```

Comment:
cc @filipeom, what do you think about this potential change in smtml?

Reply:
We have this https://github.com/formalsec/smtml/blob/67ef31468fb2251071365cbb97800ad6ad08a963/src/solver_intf.ml#L64. But internally it unfold the set into a list. You can problably using this function and we optimize it directly in smtml if we can

Reply:
Oh, great!